lib/grpc/client/adapters/mint/stream_response_process.ex

```diff
@@ -0,0 +1,178 @@
+defmodule GRPC.Client.Adapters.Mint.StreamResponseProcess do
+  @moduledoc """
+  This module represents the process responsible for consuming the
+    incoming messages from a connection. For each request, there will be
+    a process responsible for consuming its messages. At the end of a stream
+    this process will automatically be killed.
+  """
+
+  @typep accepted_types :: :data | :trailers | :headers | :error
+  @typep data_types :: binary() | Mint.Types.headers() | Mint.Types.error()
+
+  @accepted_types [:data, :trailers, :headers, :error]
+  @header_types [:headers, :trailers]
+
+  use GenServer
+
+  @spec start_link(GRPC.Client.Stream.t(), send_headers_or_trailers? :: boolean()) ::
+          GenServer.on_start()
+  def start_link(stream, send_headers_or_trailers?) do
+    GenServer.start_link(__MODULE__, {stream, send_headers_or_trailers?})
+  end
+
+  @doc """
+  Given a pid from this process, build an Elixir.Stream that will consume the accumulated
+    data inside this process
+  """
+  @spec build_stream(pid()) :: Enumerable.t()
+  def build_stream(pid) do
+    Elixir.Stream.unfold(pid, fn pid ->
```

Comment:
```suggestion
    Stream.unfold(pid, fn pid ->
```

There's no alias shadowing the module name